From a0f7c04a4912e90c2717c775b23ff79e02730078 Mon Sep 17 00:00:00 2001
From: Zhiwei Yang <quic_zhiwyang@quicinc.com>
Date: Wed, 10 Apr 2024 10:08:22 +0800
Subject: [PATCH] qcacmn: Add new cfg80211 wrapper for registering net device

Add a new wrapper wlan_cfg80211_register_netdevice() that calls
cfg80211_register_netdevice for Kernel 5.4+.

Signed-off-by: Zhiwei Yang <quic_zhiwyang@quicinc.com>
---
 os_if/linux/wlan_cfg80211.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os_if/linux/wlan_cfg80211.h b/os_if/linux/wlan_cfg80211.h
index 52420ed..85f4d16 100644
--- a/os_if/linux/wlan_cfg80211.h
+++ b/os_if/linux/wlan_cfg80211.h
@@ -624,7 +624,7 @@ wlan_cfg80211_nla_strscpy(char *dst, const struct nlattr *nla, size_t dstsize)
 }
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0) || \
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0) || \
 	(defined CFG80211_CHANGE_NETDEV_REGISTRATION_SEMANTICS))
 static inline int wlan_cfg80211_register_netdevice(struct net_device *dev)
 {
@@ -637,7 +637,7 @@ static inline int wlan_cfg80211_register_netdevice(struct net_device *dev)
 }
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 12, 0) || \
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 0) || \
 	(defined CFG80211_CHANGE_NETDEV_REGISTRATION_SEMANTICS))
 static inline void wlan_cfg80211_unregister_netdevice(struct net_device *dev)
 {
-- 
2.7.4

